test/logflare/billing/billing_counts_test.exs

```diff
@@ -0,0 +1,71 @@
+defmodule Logflare.Billing.BillingCountsTest do
+  use Logflare.DataCase
+  alias Logflare.Billing.BillingCounts
+
+  setup do
+    user = insert(:user)
+
+    _non_matching_user = insert(:billing_counts, inserted_at: ~U[2023-02-01 00:00:00Z])
+
+    _outside_boundary = insert(:billing_counts, user: user, inserted_at: ~U[2023-01-31 00:00:00Z])
+
+    _outside_boundary = insert(:billing_counts, user: user, inserted_at: ~U[2023-03-01 00:00:00Z])
+
+    inside_boundary_0 = insert(:billing_counts, user: user, inserted_at: ~U[2023-02-17 00:00:00Z])
+
+    inside_boundary_1 = insert(:billing_counts, user: user, inserted_at: ~U[2023-02-17 00:00:00Z])
+
+    inside_boundary_2 = insert(:billing_counts, user: user, inserted_at: ~U[2023-02-10 00:00:00Z])
+
+    %{
+      user: user,
+      inside_boundary_0: inside_boundary_0,
+      inside_boundary_1: inside_boundary_1,
+      inside_boundary_2: inside_boundary_2
+    }
+  end
+
+  describe "timeseries/3" do
+    test "fetches billing counts for a certain period in a time series format", %{
+      user: user,
+      inside_boundary_0: inside_boundary_0,
+      inside_boundary_1: inside_boundary_1,
+      inside_boundary_2: inside_boundary_2
+    } do
+      result = BillingCounts.timeseries(user, ~U[2023-02-01 00:00:00Z], ~U[2023-02-28 00:00:00Z])
+
+      days =
+        Date.range(~D[2023-02-01], ~D[2023-02-28]) |> Enum.map(&DateTime.new!(&1, ~T[00:00:00]))
+
+      assert Enum.all?(result, fn [d | _] ->
```

Comment:
this block of assertions are a little too complex and hard to read.

if we have 1 out and 2 in, then we just need to assert that there is 2 dates returned with a sum of 3 counts.

Reply:
I want to ensure that we're capturing the right date boundaries to avoid regressions